OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -103,7 +105,7 @@ public void Damaged(Actor self, AttackInfo e)
 			}
 
 			// not a lot we can do about things we can't hurt... although maybe we should automatically run away?
-			if (!attack.HasAnyValidWeapons(Target.FromActor(attacker)))
+			if (!attackBases.Any(Exts.IsTraitEnabled) || !attackBases.First(Exts.IsTraitEnabled).HasAnyValidWeapons(Target.FromActor(attacker)))
```

Comment:
I don't like this recent trend of "upgrading" from "requires only one" to "supports many, but breaks in unexpected ways if you have multiple active at the same time".  Can we please check over all active traits and *never* use First/FirstOrDefault?